etfl/core/memodel.py

```diff
@@ -259,24 +259,54 @@ def add_nucleotide_sequences(self, sequences):
         :param sequences:
         :return:
         """
-
+        
         for gene_id, seq in sequences.items():
             if gene_id in self.genes:
                 new = replace_by_me_gene(self, gene_id, seq)
-
+                
             else:
                 self.logger.warning('Model has no gene {}, Adding it'.format(gene_id))
                 new = ExpressedGene(id= gene_id, name = gene_id, sequence=seq)
                 self.add_genes([new])
 
             self._make_peptide_from_gene(gene_id)
+            
+            
+    def add_transcription_by(self, transcription_dict):
+        
+        for gene_id, transcripted_by in transcription_dict.items():
+            # transcripted_by is a list of rnap(s)
+            try:
+                self.genes.get_by_id(gene_id).transcribed_by = transcripted_by
+            except KeyError:
+                continue
+            
+    def add_translation_by(self, translation_dict):
+        
+        for gene_id, translated_by in translation_dict.items():
+            # transcripted_by is a list of rnap(s)
+            try:
+                self.genes.get_by_id(gene_id).translated_by = translated_by
+            except KeyError:
+                continue
 
     def _make_peptide_from_gene(self, gene_id):
         free_pep = Peptide(id=gene_id,
                            name='Peptide, {}'.format(gene_id),
                            gene_id=gene_id)
         free_pep._model = self
         self.peptides += [free_pep]
+    
+    def add_peptide_sequences(self, aa_sequences):
```

Comment:
Can you remind me for which case you ahd to add this ?

Reply:
This is a function that I have to add my peptide sequences. It's not related to mitochondrial expression. You can neglect it.